### PR TITLE
handle pastmidnight vjs

### PR DESCRIPTION
**Pull request by antoine-de**
past-midnight were not handled in kirin, creating weird delays (spanning on 4 days for my example).

We need to clear the kirin db with those weird delays

### CODE REVIEW

File: kirin/core/handler.py [hunk 1]
```diff
@@ -209,13 +212,17 @@ def merge(navitia_vj, db_trip_update, new_trip_update):
...
 
-        departure = arrival = None
-        if nav_departure_time:
-            departure = _get_datetime(new_trip_update.vj.circulation_date, nav_departure_time, timezone)
+        arrival = departure = None
         if nav_arrival_time:
-            arrival = _get_datetime(new_trip_update.vj.circulation_date, nav_arrival_time, timezone)
-
-        #TODO handle past midnight
+            if last_nav_dep and last_nav_dep > nav_arrival_time:
```
Comment (pbougue) [inline kirin/core/handler.py]: Isn't there any way to manage vj longer than 24h ?

**Comment by pbougue** [comment]
Seems OK to me.
Maybe add a test on pass midnight between 2 stops... But the code is almost the same.

**Comment by antoine-de** [comment]
yes I wondered about this as well but the code would be copy/pasted

as you wish

File: kirin/core/handler.py [hunk 2]
```diff
@@ -209,13 +212,17 @@ def merge(navitia_vj, db_trip_update, new_trip_update):
...
+                # last departure is after arrival, it's a past-midnight
+                circulation_date += timedelta(days=1)
+            arrival = _get_datetime(circulation_date, nav_arrival_time, timezone)
+        if nav_departure_time:
+            if nav_arrival_time and nav_arrival_time > nav_departure_time:
```
Comment (pbougue) [inline kirin/core/handler.py]: You can have `None` in `nav_arrival_time`... In which case I don't know how it will be handled.

Comment (xlqian) [inline kirin/core/handler.py]: @pbougue `nav_arrival_time` is actually tested before the comparison

**Pull request merged**